spec/models/manageiq/providers/kubernetes/container_manager/scanning/job_spec.rb

```diff
@@ -144,6 +144,20 @@ def fetch_metadata(*_args)
       end
     end
 
+    it 'should add correct environment variables' do
+      att_name = 'http_proxy'
+      my_value = "MY_TEST_VALUE"
+      @ems.custom_attributes.create(:section => described_class::ATTRIBUTE_SECTION,
+                                    :name    => att_name,
+                                    :value   => my_value)
+      allow_any_instance_of(described_class).to receive_messages(:kubernetes_client => MockKubeClient.new)
```

Comment:
@enoodle it sounds like the right work around here is to mock a specific object of type `described_class` instead of asserting that any instance of `described_class` can respond accordingly.

But, I just looked and this spec has `allow_any_instance_of` used all over.

Instead of nitpicking for this PR, can someone file a separate PR that cleans up the `allow_any_instance_of` usage later?